src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OpenIDAuthentication.java

```diff
@@ -0,0 +1,46 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import java.io.IOException;
+
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import fish.payara.security.annotations.LogoutDefinition;
+import fish.payara.security.annotations.OpenIdAuthenticationDefinition;
+import fish.payara.security.openid.api.OpenIdConstant;
+import jakarta.annotation.security.DeclareRoles;
+import jakarta.ejb.EJB;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.HttpConstraint;
+import jakarta.servlet.annotation.ServletSecurity;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * OIDC login implementation
+ */
+@WebServlet("/oidc/login")
+@OpenIdAuthenticationDefinition(
```

Comment:
If I understood correctly, the OIDC implementation in Payara will lookup JWKS to verify any tokens coming in. I have questions about the capabilities of this implementation:

A) How do we determine from a non-JWT bearer token which provider shall be used? Is that at all supported? We should not just assume that the access token will be a JWT, as far as I know the OIDC spec does not require that token to be a JWT. We should not make people send us ID tokens, which are defined to be JWT by the standard, as it is considered bad practice.
B) If access token coming in via the API are no JWTs, how does the OIDC implementation in Payara determine which provider to use? Do we need to add special handler logic for this by ourselves?
C) When verifying the Bearer token, the implementation retrieves the JWKS from the IdP or tries to receive an ID token from the IdPs Userinfo endpoint. Is any of this cached? What is the potential performance impact on the API?
D) Does the implementation cache the JWKS, so if the bearer token is a JWT it can verify it without extra roundtrips?

Reply:
I only have found limited documentation on this. Most things I know are either from here: https://docs.payara.fish/enterprise/docs/Technical%20Documentation/Public%20API/OpenID%20Connect%20Support.html , or from things I noticed while playing with it.